config/impl-base/src/main/java/org/jboss/arquillian/config/impl/extension/StringPropertyReplacer.java

```diff
@@ -235,47 +194,155 @@ else if (c == '}' && state == IN_BRACKET) {
         return buffer.toString();
     }
 
-    /**
-     * Try to resolve a "key" from the provided properties by
-     * checking if it is actually a "key1,key2", in which case
-     * try first "key1", then "key2". If all fails, return null.
-     * <p>
-     * It also accepts "key1," and ",key2".
-     *
-     * @param key
-     *     the key to resolve
-     * @param props
-     *     the properties to use
-     *
-     * @return the resolved key or null
-     */
-    private static String resolveCompositeKey(String key, Properties props) {
-        String value = null;
-
-        // Look for the comma
-        int comma = key.indexOf(',');
-        if (comma > -1) {
-            // If we have a first part, try resolve it
-            if (comma > 0) {
-                // Check the first part
-                String key1 = key.substring(0, comma);
-                if (props != null) {
-                    value = props.getProperty(key1);
-                } else {
-                    value = System.getProperty(key1);
+
+    public interface PropertyResolver {
+        String getValue(String key);
+    }
+
+    static class ClasspathPropertyResolver implements PropertyResolver {
+
+        /**
+         * Classpath base property
+         */
+        private static final String CLASSPATH = "classpath(";
+
+        public String getValue(String key) {
+
+            if (key.startsWith(CLASSPATH)) {
+                final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+                String classpathResource = key.substring(CLASSPATH.length(), key.length() - 1);
+                final URL resource = contextClassLoader.getResource(classpathResource);
+
+                //If resource is not found it is returned as null so no change is applicable.
```

Comment:
It would be also good to log some information that the resource wasn't loaded from classpath

Reply:
Notice that in case of system properties if they are not resolved then no log is printed, they are just returned as is, so why we should do in case of classpath? What I mean is that if we do not log in case of sysprops nor envvars then it seems reasonable to not do it in case of classpath.

Reply:
sysprops is something expected and obvious - it works in bash, pom files and other programming languages in the same way. Resolving `${classpath()}` is something completely special and new just for our Arquillian case... That's why I would make it transparent what is going on and why it wasn't resolved...